source/features/bypass-checks.tsx

```diff
@@ -1,19 +1,20 @@
+import mem from 'mem';
 import onetime from 'onetime';
 import {observe} from 'selector-observer';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
-import fetchDom from '../helpers/fetch-dom';
+import * as api from '../github-helpers/api';
+import {getRepositoryInfo} from '../github-helpers';
 
-async function bypass(detailsLink: HTMLAnchorElement): Promise<void> {
-	const directLink = await fetchDom<HTMLAnchorElement>(
-		detailsLink.href,
-		'[data-hydro-click*="check_suite.external_click"]'
-	);
+const getDirectLink = mem(async (runNumber: number): Promise<string> => {
+	const directLink = await api.v3(`https://api.github.com/repos/${getRepositoryInfo()!.owner!}/${getRepositoryInfo()!.name!}/check-runs/${runNumber}`);
```

Comment:
I think it's worth using v4 so that we can also limit the amount of requests: https://developer.github.com/v4/object/checkrun/#detailsurl

Reply:
I know however that this complicates it considerably with `selector-observer` since you'd have to build a list in `add`, then in the next tick, make the joint v4 query.

Reply:
Unless you can figure out how to use that thing, I gave up!

It was a nest of 7 and was only per commit. In other words I could not search for a specific run I was only able to load a commit and then load all its _runs_.

V3 we can search for a specific run

Reply:
@sindresorhus if you respond in time: what's a good pattern for this? It's like the `event loop` but instead of callbacks it's a list of elements

1. Add to array in loop
2. Wait for the next tick (likely via `await Promise.resolve()`?)
3. Use the array

Reply:
I had missed your last comment. Thanks for trying. It sucks that the GraphQL API is so limited compared to REST.